### PR TITLE
Add Go snippets for file attachments

### PR DESCRIPTION
**Pull request by philkra**
Closes #

## Summary

follow up of #197 

### Timing

**Release**:

- [ ] When ready
- [ ] Hold for release | Release after: <mm/dd/yy>

### CODE REVIEW

File: 040-SDK/150-file-attachments.mdx [hunk 1]
```diff
@@ -369,7 +452,7 @@ photos = xata.data().query("Users", {
 
 The `base64Content` and ` signedUrl` will only be returned when requested explicitly. They are not included by default when requesting all fields using wildcard.
```
Comment (joan-ing) [inline 040-SDK/150-file-attachments.mdx:453]:
```suggestion
Fields `base64Content` and `signedUrl` within the `photo` object are retrieved only when they are explicitly specified in the request. If you use a wildcard to request all fields of the photo object, these specific fields will not be included automatically. You must list them individually to receive their data.
```

File: 040-SDK/150-file-attachments.mdx [hunk 2]
```diff
@@ -385,6 +468,20 @@ photos = xata.data().query("Users", {
 })
 ```
 
+```go
+searchClient, _ := xata.NewSearchAndFilterClient()
+records, _ := searchClient.Query(context.TODO(), xata.QueryTableRequest{
+  TableName: "Users",
+  Payload: xata.QueryTableRequestPayload{
+    Columns: []string{"name", "photo.base64Content", "photo.signedUrl", "photo.*"},
+    // Nested filters currently are not support yet in xata-go
```
Comment (joan-ing) [inline 040-SDK/150-file-attachments.mdx:477]:
```suggestion
    // Nested filters are currently not supported in xata-go
```

File: 030-Concepts/075-file-attachments.mdx [hunk 3]
```diff
@@ -123,19 +153,17 @@ Within the Xata UI you can make files public by default for the column by settin
 ![Entire columns can be made public by default](images/column-access-public.png)
 
 <Alert status="warning">
-  **Public object caching**
-When a public object gets cached, changing permissions will not invalidate the cached entry.
-The file is still accessible through old URLs until the cache entry expires in 2h. Xata advises greatest caution when
-configuring public access, because in practice there is a delay in changing the permissions back from public to
-private.
-
+  **Public object caching** When a public object gets cached, changing permissions will not invalidate the cached entry.
```
Comment (joan-ing) [inline 030-Concepts/075-file-attachments.mdx:156]:
```suggestion
  **Public object caching** Changing the permissions of a cached public object does not remove it from the cache; the cached version remains unaffected.
```

File: 030-Concepts/075-file-attachments.mdx [hunk 4]
```diff
@@ -123,19 +153,17 @@ Within the Xata UI you can make files public by default for the column by settin
...
+  The file is still accessible through old URLs until the cache entry expires in 2h. Xata advises greatest caution when
```
Comment (joan-ing) [inline 030-Concepts/075-file-attachments.mdx:157]:
```suggestion
  The file remains accessible through the previously existing URLs until the cache expires after 2 hours. It is highly recommended to exercise caution when
```

File: 030-Concepts/075-file-attachments.mdx [hunk 5]
```diff
@@ -123,19 +153,17 @@ Within the Xata UI you can make files public by default for the column by settin
...
+  configuring public access, because in practice there is a delay in changing the permissions back from public to
```
Comment (joan-ing) [inline 030-Concepts/075-file-attachments.mdx:158]:
```suggestion
  configuring public access. There is a significant delay in reverting permissions from public back to 
```

**Review by joan-ing — APPROVED** [review]
lgtm. Just a couple minor tweaks

**Comment by philkra** [comment]
Thank you @joan-ing , the suggestions are committed.

**Pull request merged**